src/Rings/mpoly-localizations.jl

```diff
@@ -3018,14 +3018,32 @@ true
 ### Some auxiliary functions
 
 @attr MPolyLocalizedIdeal function radical(I::MPolyLocalizedIdeal)
-  J = pre_saturated_ideal(I)
-  return ideal(base_ring(I), gens(radical(J)))
+  has_attribute(I, :is_prime) && get_attribute(I, :is_prime) && return I
+  # heuristics have shown that the radical of the saturated ideal 
+  # is often quicker 
+  #J = pre_saturated_ideal(I)
+  J = saturated_ideal(I)
+  R = base_ring(I)
+  return ideal(base_ring(I), [g for g in R.(gens(radical(J))) if !is_zero(g)])
 end
 
-@attr function dim(I::MPolyLocalizedIdeal)
+@attr function dim(I::MPolyLocalizedIdeal{RT}) where {RT<:MPolyLocRing{<:Field, <:FieldElem, <:MPolyRing, <:MPolyRingElem, <:MPolyPowersOfElement}}
   return dim(saturated_ideal(I))
 end
 
+@attr function dim(I::MPolyLocalizedIdeal{RT}) where {RT<:MPolyLocRing{<:Field, <:FieldElem, <:MPolyRing, <:MPolyRingElem, <:MPolyComplementOfPrimeIdeal}}
+  return dim(saturated_ideal(I)) - dim(prime_ideal(inverted_set(base_ring(I))))
+end
+
+@attr function dim(I::MPolyLocalizedIdeal{RT}) where {RT<:MPolyLocRing{<:Field, <:FieldElem, <:MPolyRing, <:MPolyRingElem, <:MPolyComplementOfKPointIdeal}}
+  J = shifted_ideal(I)
+  # TODO: Is there a more conceptual way to do this???
+  oo = negdegrevlex(gens(base_ring(J)))
+  standard_basis(J, ordering=oo) # result is cached, but not returned
+  return Singular.dimension(J.gb[oo].gens.S)
```

Comment:
@ederc : Do you have any suggestions how to address this todo?

Reply:
@HechtiDerLachs : I would go for dimension(leading_ideal(J, ordering=oo))
(The leading ideal lives by definition in the polynomial ring and hence, we can use the usual dimension on it. And a theorem states that the dimension of the leading ideal is the dimension of the respective ideal.)